app/lib/submission_builder/ty_2021/adv_ctc_irs1040.rb

```diff
@@ -42,33 +42,42 @@ def document
-            xml.TotalItemizedOrStandardDedAmt tax_return.standard_deduction # 12
+            xml.TotalItemizedOrStandardDedAmt tax_return.standard_deduction # line 12a in 2021v5.2/IndividualIncomeTax/Ind1040/IRS1040/IRS1040.xsd
+            xml.TotDedCharitableContriAmt tax_return.standard_deduction # 12c
+            xml.TotalDeductionsAmt tax_return.standard_deduction # 14
             xml.TaxableIncomeAmt 0 # 15
+
+            # Line 28: remaining amount of CTC they are claiming (as determined in flow and listed on 8812 14i
+            claimed_child_tax_credit = "0" # TODO: eventually replace this amount with the adv_ctc_amount_received when this is set in the RRC stories
+            xml.RefundableCTCOrACTCAmt claimed_child_tax_credit # 28
+            # maybe add this to be part of the stories for RRC
+
+            # Line 30: remaining amount of RRC they are claiming (as determined in flow). If they are not claiming anything, the Revenue Procedure is ambiguous as to whether this should be $0 or blank; but I recall last year having an issue with blank, so I would advocate $0. Rev Proc is clear IRS will not correct $0.
             xml.RecoveryRebateCreditAmt tax_return.claimed_recovery_rebate_credit # 30
-            xml.RefundableCreditsAmt tax_return.claimed_recovery_rebate_credit # 32
-            xml.TotalPaymentsAmt tax_return.claimed_recovery_rebate_credit # 33
-            xml.OverpaidAmt tax_return.claimed_recovery_rebate_credit # 34
-            xml.RefundAmt tax_return.claimed_recovery_rebate_credit # 35a
+
+            # Line 32, 33, 34, 35a: Line 28 + Line 30
+            total_payments = (claimed_child_tax_credit.to_i + tax_return.claimed_recovery_rebate_credit.to_i).to_s
+            xml.RefundableCreditsAmt total_payments # 32
+            xml.TotalPaymentsAmt total_payments # 33
+            xml.OverpaidAmt total_payments # 34
+            xml.RefundAmt total_payments # 35a
+
             if bank_account.present? && intake.refund_payment_method_direct_deposit?
-              xml.RoutingTransitNum account_number_type(bank_account.routing_number)
-              xml.BankAccountTypeCd bank_account.account_type_code
-              xml.DepositorAccountNum account_number_type(bank_account.account_number)
+              xml.RoutingTransitNum account_number_type(bank_account.routing_number) # 35b
+              xml.BankAccountTypeCd bank_account.account_type_code # 35c
+              xml.DepositorAccountNum account_number_type(bank_account.account_number) # 35d
             end
             xml.RefundProductCd "NO FINANCIAL PRODUCT"
           }
         end.doc
       end
 
       def build
-        unless ENV['TEST_SCHEMA_VALIDITY_ONLY'] == 'true'
-          raise NotImplementedError, "SubmissionBuilder::TY2021::AdvCtcIrs1040 is for testing purposes only and does not currently conform to the 2021 revenue procedure."
-        end
-
+        # unless ENV['TEST_SCHEMA_VALIDITY_ONLY'] == 'true'
```

Comment:
i think you can fully remove this commented out code.

Reply:
hmm actually you know what... we've still got a lot of placeholders here. i think what we actually need to do is add an around filter to the tests to set the env variable and revert to having this warning show up if you try to run the code directly. that way we don't forget that there are things that need to be updated in here.